gdbplus/gdb-12.1/gdb/gdb_dep.c

```diff
@@ -2797,3 +2797,114 @@ ca_eval_address(const char* expr)
 {
 	return parse_and_eval_address(expr);
 }
+
+// return true if an unamed union has a data member of given name
+static bool
+union_has_field_name(struct type *type, const char *field_name)
+{
+	type = check_typedef (type);
+	if (type->code() != TYPE_CODE_UNION)
+		return false;
+	for (int i = 0; i < type->num_fields(); i++) {
+		const char *name = type->field(i).name();
+		if (name && *name != '\0') {
+			if (strcmp (field_name, name) == 0)
+				return true;
+		} else if (union_has_field_name(type->field(i).type(), field_name)) {
+			return true;
+		}
+	}
+	return false;
+}
+
+static int
+type_field_name2no(struct type *type, const char *field_name)
```

Comment:
it might be better to use `std::optional` to express the idea that sometimes there is no number instead of using -1 for the missing case.

Reply:
reference: https://en.cppreference.com/w/cpp/utility/optional

Reply:
cool